autolens/lens/ray_tracing.py

```diff
@@ -592,7 +592,10 @@ def traced_sparse_grids_of_planes_from_grid(
                 )
                 traced_sparse_grids_of_planes.append(traced_sparse_grids[plane_index])
 
-        return traced_sparse_grids_of_planes
+        if len(sparse_image_plane_grids_of_planes) > 1:
+            return traced_sparse_grids_of_planes, sparse_image_plane_grids_of_planes[1]

```

Comment:
This would be more concise if you used the if-else to compute an index

grid_index = 1 if len(sparse_image_plane_grids_of_plane) > 1 else 0

Reply:
The changes to the ray_tracing module are horrific. It needs a good 'ol refactor but I dont have time to do it properly atm.